server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -152,6 +157,52 @@
 public class CompactionCoordinator
     implements CompactionCoordinatorService.Iface, Runnable, MetricsProducer {
 
+  // Object that serves as a TopN view of the RunningCompactions, ordered by
+  // RunningCompaction start time. The first entry in this Set should be the
+  // oldest RunningCompaction.
+  private static class TimeOrderedRunningCompactionSet
```

Comment:
Would probably be better to wrap ConcurrentSkipListSet rather than extend it because it has lots of methods for adding and removing that are not covered like addAll, removeIf, removeAll, etc.  If its wrapped none of the unsupported method are available.